### PR TITLE
[2201.3.x] Add langlib examples for decimal

### LINKED ISSUE #38911: [Improvement]: Add lang library examples for decimal langlib apidocs
### Description

Currently lang library API docs does not contains usage examples.
So we need to add examples for decimal lang library API docs.

### Describe your problem(s)

_No response_

### Describe your solution(s)

_No response_

### Related area

-> Other Area

### Related issue(s) (optional)

_No response_

### Suggested label(s) (optional)

_No response_

### Suggested assignee(s) (optional)

_No response_

### LINKED ISSUE #39090: [Improvement]: Add lang library examples for decimal langlib apidocs in 2201.3.x
### Description

Currently lang library API docs does not contains usage examples.
So we need to add examples for decimal lang library API docs.

### Describe your problem(s)

_No response_

### Describe your solution(s)

_No response_

### Related area

-> Other Area

### Related issue(s) (optional)

_No response_

### Suggested label(s) (optional)

_No response_

### Suggested assignee(s) (optional)

_No response_

### PR DESCRIPTION
**Pull request by SasinduDilshara**
Fixes #38911

### CODE REVIEW

File: langlib/lang.decimal/src/main/ballerina/decimal.bal [hunk 1]
```diff
@@ -66,6 +101,17 @@ public isolated function abs(decimal x) returns decimal = @java:Method {
 # roundToIntegralTiesToEven will return a value with a positive exponent when the operand has a positive exponent.
 # Note that `<int>x` is the same as `<int>x.round(0)`.
 #
+# ```ballerina
+# decimal d = 3.55;
```
Comment (pcnfernando) [inline langlib/lang.decimal/src/main/ballerina/decimal.bal:105]: Shall we add two cases to show half even mode?
eg: 2.5 to 2 and 3.5 to 4

File: langlib/lang.decimal/src/main/ballerina/decimal.bal [hunk 2]
```diff
@@ -110,6 +176,14 @@ public isolated function ceiling(decimal x) returns decimal = @java:Method {
 # - a FloatingPointTypeSuffix is not allowed
 # This is the inverse of function ``value:toString`` applied to an `decimal`.
 #
+# ```ballerina
+# decimal:fromString("0.2453") ⇒ 0.2453
+# 
+# decimal:fromString("-10") ⇒ -10
+# 
+# decimal:fromString("") ⇒ error
```
Comment (pcnfernando) [inline langlib/lang.decimal/src/main/ballerina/decimal.bal:184]: Shall we use an actual erroneous string instead of empty str?

**Comment by pcnfernando** [comment]
@SasinduDilshara intermittent debugger failure should have been fixed in upstream branches. Try resyncing

File: langlib/lang.decimal/src/main/ballerina/decimal.bal (hunk 2 above)
Comment (pcnfernando) [reply langlib/lang.decimal/src/main/ballerina/decimal.bal:184]: can you share the review comment

Comment (SasinduDilshara) [reply langlib/lang.decimal/src/main/ballerina/decimal.bal:184]: https://github.com/ballerina-platform/ballerina-lang/pull/38905#discussion_r1049245698

Comment (pcnfernando) [reply langlib/lang.decimal/src/main/ballerina/decimal.bal:184]: I was thinking
```ballerina
# decimal:fromString("123d") ⇒ error
```
@MaryamZi wdyt?

Comment (MaryamZi) [reply langlib/lang.decimal/src/main/ballerina/decimal.bal:184]: +1, as long as it is not an obviously non-parseable string.

Comment (SasinduDilshara) [reply langlib/lang.decimal/src/main/ballerina/decimal.bal:184]: Ok, will update

File: langlib/lang.decimal/src/main/ballerina/decimal.bal (hunk 1 above)
Comment (SasinduDilshara) [reply langlib/lang.decimal/src/main/ballerina/decimal.bal:105]: Updated

File: langlib/lang.decimal/src/main/ballerina/decimal.bal (hunk 2 above)
Comment (SasinduDilshara) [reply langlib/lang.decimal/src/main/ballerina/decimal.bal]: Updated

File: langlib/lang.decimal/src/main/ballerina/decimal.bal [hunk 3]
```diff
@@ -74,8 +123,18 @@ public isolated function round(decimal x, int fractionDigits = 0) returns decima
...
+# Return a decimal with a specified value and exponent.
+# Return a decimal value that has the same value (except for rounding) as the first
+# argument, and the same exponent as the second argument.
+# This is the IEEE quantize operation.
+#
+# ```ballerina
+# decimal d = 4.16;
+# d.quantize(3.512) ⇒ 4.160
+# 
+# decimal:quantize(3.15, 3.512) ⇒ 3.150
```
Comment (MaryamZi) [inline langlib/lang.decimal/src/main/ballerina/decimal.bal:135]: See https://github.com/ballerina-platform/ballerina-lang/pull/38920#discussion_r1052152007

Comment (MaryamZi) [reply langlib/lang.decimal/src/main/ballerina/decimal.bal]: https://github.com/ballerina-platform/ballerina-lang/pull/38920#discussion_r1052966534

**Review by dulajdilshan — APPROVED** [review]
LGTM!

File: langlib/lang.decimal/src/main/ballerina/decimal.bal (hunk 3 above)
Comment (SasinduDilshara) [reply langlib/lang.decimal/src/main/ballerina/decimal.bal]: Updated. Please check

**Review by MaryamZi — CHANGES_REQUESTED** [review]
See comments on https://github.com/ballerina-platform/ballerina-lang/pull/38920.

**Review by mohanvive — APPROVED** [review]

**Review by MaryamZi — APPROVED** [review]

**Review by pcnfernando — APPROVED** [review]

**Pull request merged**